### PR TITLE
MGMT-9561: VSphere platform lacking setting disk.EnableUUID guidance

### PR DESCRIPTION
**Pull request by michaellevy101**

### CODE REVIEW

File: src/inventory/disks.go [hunk 1]
```diff
@@ -124,6 +124,26 @@ func (d *disks) getBootable(path string) bool {
...
+func (d *disks) hasUUID(path string) bool {
+	if path == "" {
+		return false
+	}
+
+	// Openshift requires VSphere (and maybe other platforms) VMs to have a UUID for disks
+	// Here we add this information to the inventory for the assisted-service verification.
+	// Getting device page 83 which contains the device WWID(disk.UUID)
+	// please see: https://access.redhat.com/solutions/93943
+	stdout, stderr, exitCode := d.dependencies.Execute("sg_inq", "-p", "0x83", path)
```
Comment (omertuc) [inline src/inventory/disks.go:136]:
```
$ podman run -it --entrypoint bash quay.io/edge-infrastructure/assisted-installer-agent:latest -c "sg_inq"

bash: sg_inq: command not found
```

File: src/inventory/disks.go [hunk 2]
```diff
@@ -124,6 +124,26 @@ func (d *disks) getBootable(path string) bool {
...
+
+	if exitCode != 0 {
+		logrus.Warnf("Could not get UUID information for path %s: %s", path, stderr)
```
Comment (omertuc) [inline src/inventory/disks.go:139]: I'm afraid users will see this and get scared. Can you rephrase it to something less alerting, maybe something like:

```suggestion
		logrus.Infof("hasUUID is false for path %s: %s", path, stderr)
```

Comment (michaellevy101) [reply src/inventory/disks.go:139]: hasUUID is false isn't clear enough. 
Updated to info

Comment (omertuc) [reply src/inventory/disks.go]: >hasUUID is false isn't clear enough.

I know, it's intentionally vague to make naive users ignore it.

Sadly our users don't care about the severity of the messages. They look for error-looking messages everywhere (even inside the inventory itself, and constantly post them on our Slack channel and open BZs about them.

If the user has even 1 disk that triggers this , then they're gonna see this messages show up in the agent logs every 60 seconds (everytime there's an inventory step), it's gonna look scary and they're gonna waste their time on it, report it as a bug, or ask about it on Slack, wasting even more people's time.

I'm not just speculating, this is how our users behave, they look for error-looking strings *[everywhere](https://issues.redhat.com/browse/MGMT-7388)*.

File: src/inventory/disks.go [hunk 3]
```diff
@@ -124,6 +124,26 @@ func (d *disks) getBootable(path string) bool {
...
+	if path == "" {
+		return false
+	}
+
+	// Openshift requires VSphere (and maybe other platforms) VMs to have a UUID for disks
+	// Here we add this information to the inventory for the assisted-service verification.
+	// Getting device page 83 which contains the device WWID(disk.UUID)
+	// please see: https://access.redhat.com/solutions/93943
+	stdout, stderr, exitCode := d.dependencies.Execute("sg_inq", "-p", "0x83", path)
+
```
Comment (romfreiman) [inline src/inventory/disks.go:137]: maybe worth writing here the output?

File: src/inventory/disks.go (hunk 2 above)
Comment (michaellevy101) [reply src/inventory/disks.go]: thanks for clarifying

File: src/inventory/disks.go [hunk 4]
```diff
@@ -124,6 +124,27 @@ func (d *disks) getBootable(path string) bool {
...
+func (d *disks) hasUUID(path string) bool {
+	if path == "" {
+		return false
+	}
+
+	// Openshift requires VSphere (and maybe other platforms) VMs to have a UUID for disks
+	// Here we add this information to the inventory for the assisted-service verification.
+	// Getting device page 83 which contains the device WWID(disk.UUID)
+	// please see: https://access.redhat.com/solutions/93943
+	stdout, stderr, exitCode := d.dependencies.Execute("sg_inq", "-p", "0x83", path)
```
Comment (tsorya) [inline src/inventory/disks.go:136]: will it work for arm? this package exists on it?

File: Dockerfile.assisted_installer_agent [hunk 5]
```diff
@@ -19,6 +19,7 @@ RUN dnf install -y \
 		# inventory
 		dmidecode ipmitool biosdevname file fio \
 		smartmontools \
+		sg3_utils \
```
Comment (tsorya) [inline Dockerfile.assisted_installer_agent:22]: this will require updating downstream images

Comment (romfreiman) [reply Dockerfile.assisted_installer_agent:22]: when?

Comment (osherdp) [reply Dockerfile.assisted_installer_agent:22]: We can update D/S image before this PR gets in

File: src/inventory/disks.go (hunk 4 above)
Comment (michaellevy101) [reply src/inventory/disks.go:136]: @danielerez ?

Comment (danielerez) [reply src/inventory/disks.go:136]: I guess it would be simplest to verify it on an arm64 machine, i.e. try to install/run on the machine.

Comment (danielerez) [reply src/inventory/disks.go:136]: Seems the pkg is available for arm: https://centos.pkgs.org/7/centos-aarch64/sg3_utils-1.37-19.el7.aarch64.rpm.html

**Comment by michaellevy101** [comment]
/?

**Comment by michaellevy101** [comment]
@osherdp @eliorerz just remove the label?

**Comment by osherdp** [comment]
Sure, I removed the label
Keep an eye on the D/S builds, to make sure all goes well

**Comment by michaellevy101** [comment]
@osherdp how did you remove it?
